src/Common/src/Interop/FreeBSD/Interop.libutil.cs

```diff
@@ -0,0 +1,429 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+
+internal static partial class Interop
+{
+    internal static partial class libutil
+    {
+        private const ulong SecondsToNanoSeconds = 1000000000;
+
+        // Constants from sys/syslimits.h
+        private const int PATH_MAX  = 1024;
+
+        // Constants from sys/user.h
+        private const int TDNAMLEN = 16;
+        private const int WMESGLEN = 8;
+        private const int LOGNAMELEN = 17;
+        private const int LOCKNAMELEN = 8;
+        private const int COMMLEN = 19;
+        private const int KI_EMULNAMELEN = 16;
+        private const int LOGINCLASSLEN = 17;
+        private const int KI_NGROUPS = 16;
+
+        private const int KI_NSPARE_INT = 4;
+        private const int KI_NSPARE_LONG = 12;
+        private const int KI_NSPARE_PTR = 6;
+
+
+        // Constants from sys/_sigset.h
+        private const int _SIG_WORDS = 4;
+
+        // Constants from sys/sysctl.h
+        private const int CTL_KERN = 1;
+        private const int KERN_PROC = 14;
+        private const int KERN_PROC_PATHNAME = 12;
+        private const int KERN_PROC_PROC = 8;
+        private const int KERN_PROC_ALL = 0; 
+        private const int KERN_PROC_PID  = 1;
+        private const int KERN_PROC_INC_THREAD = 16;
+
+        // Constants from proc_info.h
+        private const int MAXTHREADNAMESIZE = 64;
+        private const int PROC_PIDTASKALLINFO = 2;
+        private const int PROC_PIDTHREADINFO = 5;
+        private const int PROC_PIDLISTTHREADS = 6;
+        private const int PROC_PIDPATHINFO_MAXSIZE = 4 * PATH_MAX;
+
+        internal struct proc_stats
+        {
+            internal long startTime;        /* time_t */
+            internal int nice;
+            internal ulong userTime;        /* in ticks */
+            internal ulong systemTime;      /* in ticks */
+        }
+
+        // From sys/_sigset.h
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct sigset_t
+        {
+            fixed int bits[4];
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct uid_t
```

Comment:
We've tried not to bake assumptions on native type sizes into the managed code, instead preferring to make an internal contract between the managed code and our own native shims.  That's most important for code that needs to span different Unixes, so this might be fine for just FreeBSD, but we should be careful and consider going the shim route.